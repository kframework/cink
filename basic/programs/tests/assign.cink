int main() {
  int x, y, z;
  cout << (x = 2) + (y = z = 3) << endl;
  return 0;
}
