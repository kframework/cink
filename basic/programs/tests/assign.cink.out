5
<T>
    <in>
        ListItem(#buffer ( "" ))
        ListItem(#istream ( 0 ))
    </in>
    <out>
        ListItem(#ostream ( 1 ))
        ListItem(#buffer ( "" ))
    </out>
    <control>
        <k>
            .K
        </k>
        <return>
            0
        </return>
        <fstack>
            .List
        </fstack>
        <sideEffects>
            .Bag
        </sideEffects>
        <env>
            main |-> #symScalLoc(1)
        </env>
    </control>
    <genv>
        main |-> #symScalLoc(1)
    </genv>
    <store>
        #symScalLoc(1) |-> lambda (  @ int x ; int y ; int z ; ((cout << (
          (x = 2) + (y = (z = 3)))) << (endl)) ; return 0 ; )
        #symScalLoc(2) |-> 2
        #symScalLoc(3) |-> 3
        #symScalLoc(4) |-> 3
    </store>
</T>
